wordpress-mobile/WordPress-iOS
# Comments: show Trashed and Spam

Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/15955
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/358

When a filter is selected, its `NSPredicate` is sent back to `CommentsViewController`. It is then used when re-fetching Comments. This allows Trashed and Spam comments to be displayed correctly.

Outstanding issues:
- The results are still sorted by status, then date. 
- The selected filter is not yet saved.

To test:

- Go to Site > Comments.
- On the web, go to `Manage Comments` for the same site. (https://wordpress.com/comments/all/yoursite)
- In the app:
  - Verify all the filters show the same Comments as the web.
  - Verify only Pending comments have the yellow dot.
  - Verify the param log messages no longer appear. (`🔴 REST > params`, `🔴 XMLRPC > params`)
- Disable `commentFilters` feature flag.  
  - Verify the filter tab bar is not displayed, and only Pending and Approved comments are shown as before.

PR submission checklist:
